File: Luna.Models.Tasks.View/CardAttributes/TypeView.cs
using Luna.Models.Tasks.Domain.CardAttributes;

namespace Luna.Models.Tasks.View.CardAttributes;

public class TypeView
{
	public Guid Id { get; set; }

	public String Name { get; set; }

	public String Color { get; set; }

	public TypeView(TypeDomain typeDomain)
	{
		Id = typeDomain.Id;
		Name = typeDomain.Name;
		Color = typeDomain.Color;
	}
}

File: luna/src/models/card/typeView.ts
import {Guid} from "guid-typescript";

export default class TypeView {
    constructor(id: Guid, name: string, color: string) {
        this.id = id;
        this.name = name;
        this.color = color;
    }

    public id: Guid;
    public name: string;
    public color: string;
}